### PR TITLE
feat(desktop): make file paths in chat messages clickable

### PR DESCRIPTION
**Pull request by bdeenyy**
## Summary

Make file paths in chat messages clickable in the desktop app. A remark plugin detects Unix (`/home/user/file.rs`), tilde (`~/.config/app/settings.json`), and Windows (`C:\Users\dev\project\index.ts`) paths in markdown text and inline code, and renders them as `open-file://` links. Clicking a path reveals the file in Finder/Explorer via `shell.showItemInFolder()`.

Rebased onto latest `main`; duplicate scrollbar commit from #9601 removed.

### Testing

- `pnpm test -- src/utils/linkifyPaths.test.ts` — 14 unit tests (path detection, edge cases, trailing punctuation)
- Full desktop test suite passes locally (396 tests)
- Manual: click Unix, tilde, and Windows paths in chat messages

### Related Issues

N/A — happy to open a discussion/issue if you'd prefer to track this separately.

### Screenshots/Demos (for UX changes)

Before: file paths in chat are plain text, not clickable.

After: paths render as links; clicking opens the file location in the system file manager.

### Screenshots/Demos (for UX changes)
Before:  
image
After:   
image

### CODE REVIEW

**Comment by DOsinga** [comment]
Thanks for the contribution, and congrats on landing #9601! 🎉

We'd like to take this one but there are a few things to sort out first:

- **Codex P1 (the feature currently crashes):** In `findPaths`, the path regexes use a non-capturing prefix group `(?:...)?` plus a single capturing group for the path. That means `m[1]` is the path and `m[2]` is `undefined`. The code reads `path = m[2]` and `prefixLen = m[1].length`, so at runtime `path.length` throws `Cannot read properties of undefined` — markdown rendering will break on any message that contains a file path. Please fix the group indexing (path is `m[1]`, prefix length should come from `m[0].length - m[1].length`).
- **Tests don't test the real code:** `linkifyPaths.test.ts` re-declares the regexes and a copy of `findPaths` inline and asserts against that copy, rather than importing `findPaths` from the module. Please import and test the actual implementation — with the real (currently broken) indexing, these tests would catch the bug above.
- A couple of other open Codex comments are worth addressing too: trailing punctuation getting pulled into links (e.g. `/tmp/result.txt.`), and `~/` paths not being tilde-expanded before `shell.showItemInFolder`, so they won't resolve.

Also note that, per our [contribution guide](https://github.com/aaif-goose/goose/blob/main/CONTRIBUTING.md#discussions-issues-and-prs), for a new feature like this it's best to open an issue/discussion first so we can align on direction.

Could you address the Codex comments (fix or a one-line reason per [AI Code Reviews](https://github.com/aaif-goose/goose/blob/main/CONTRIBUTING.md#ai-code-reviews))? We'll revisit in a few days. Thanks!

File: ui/desktop/src/utils/linkifyPaths.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import { visit } from 'unist-util-visit';
+import type { Plugin } from 'unified';
+import type { Root, Text, InlineCode, Link, Parent } from 'mdast';
+
+const OPEN_FILE_PROTOCOL = 'open-file://';
+
+const UNIX_PATH_RE = /(?:^|[\s('"`[(,;]|\/\*.*?\*\/)?(\/(?:[a-zA-Z0-9._+-]+\/){1,}[a-zA-Z0-9._+-]+)/g;
+const TILDE_PATH_RE = /(?:^|[\s('"`[(,;]|\/\*.*?\*\/)?(~\/(?:[a-zA-Z0-9._+-]+\/)*[a-zA-Z0-9._+-]+)/g;
+const WIN_PATH_RE = /(?:^|[\s('"`[(,;]|\/\*.*?\*\/)?([A-Za-z]:[\\/](?:[a-zA-Z0-9._+-]+[\\/])+[a-zA-Z0-9._+-]+)/g;
```
[the remark this answers is not in the document]
Comment (bdeenyy) [reply ui/desktop/src/utils/linkifyPaths.ts]: Thanks for the review and for merging #9601!

I've addressed the feedback in e310220:

P1 — regex group indexing: findPaths now reads the path from m[1] and computes prefixLen as m[0].length - m[1].length.

Tests: linkifyPaths.test.ts imports and tests the exported findPaths from the module instead of duplicating the logic inline.

Trailing punctuation: sentence-ending punctuation (e.g. /tmp/result.txt.) is stripped before creating the link.

Tilde paths: open-path-in-explorer now calls expandTilde() before shell.showItemInFolder().

Other Codex comments:

Nested links (P2): text/inlineCode nodes inside existing link parents are skipped during traversal.
URL false positives: paths immediately after a URL scheme (https://...) are ignored.
Also rebased onto latest main and dropped the duplicate scrollbar commit (already landed in #9601).

Re: opening an issue/discussion first — understood for future features. Happy to open one retroactively if that helps track this.

Let me know if anything else needs adjusting!

[the remark this answers is not in the document]
Comment (bdeenyy) [reply ui/desktop/src/utils/linkifyPaths.ts]: Addressed the P2 "paths with spaces" feedback in 8ed41e8:

Replaced greedy regex with a segment parser that allows spaces inside folder/file names but stops before trailing prose.
open-file:// hrefs now use encodeURI() so spaces survive markdown URL handling.

File: ui/desktop/src/utils/linkifyPaths.ts [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+  let j = spaceIndex + 1;
+  while (j < text.length && isPathChar(text[j])) {
+    j++;
+  }
+  if (j === spaceIndex + 1) return spaceIndex;
+
+  const after = text[j];
+  if (after === separator) return j;
+  if (after === undefined || /[.,;:!?)'\]"]/.test(after)) return j;
```
[the remark this answers is not in the document]
Comment (bdeenyy) [reply ui/desktop/src/utils/linkifyPaths.ts]: Fixed in 692f479: spaced continuations before punctuation now require path-like evidence (capitalized/dotted segment name, or short preceding token for multi-word folder names like my documents). Trailing sentence punctuation is stripped while scanning continuations so . is not absorbed via isPathChar.

**Comment by DOsinga** [comment]
Thanks for all the work on this, and genuinely for being so responsive to the review — you fixed the P1 and the test issue quickly, and stuck with it through a lot of codex rounds.

That said, watching what codex keeps surfacing on every commit (line:col suffixes, `(1).pdf`, `[OK]` tags, apostrophes, `... for details`, years vs counts, and so on), I think this is telling us the *approach* is the problem rather than any individual bug. The current scanner starts at a separator and reads **forward** into open prose, so the one reliable landmark — the file extension — arrives late or never, and the right edge of the path has to be guessed. That guessing is an unwinnable game: every heuristic fixes one case and opens another, which is exactly the churn we are seeing. So in its current form this is not the way, and I am going to close it.

If you (or anyone) wants to take another run at it, here is a direction that sidesteps almost all of this:

- Keep a fixed **set of known extensions** (plus extensionless names like `Dockerfile`).
- Find every `.<known-ext>` occurrence and scan **backwards**, consuming allowed path chars until you hit a `/` (or `C:\` on Windows), a clear boundary, or a length cap.

Because the extension **anchors the right edge**, you never have to guess where the path ends — which is what causes 80% of the edge cases here. The only remaining judgement is how far left to go across spaces, and even there a simple default (don't cross spaces, or only cross while the previous token is path-chars-only) terminates cleanly with no per-case heuristics. That should collapse most of this ~600-line file and make it robust by construction.

Please feel free to open a fresh PR (or reopen) if you land on a more scalable solution along those lines — I'd be happy to look. Thanks again! 🙏
